source/api/commands/route2.md

```diff
@@ -0,0 +1,385 @@
+---
+title: route2
+containerClass: experimental
+---
+
+{% note warning %}
+{% fa fa-warning orange %} **This is an experimental feature. In order to use it, you must manually configure the `experimentalNetworkMocking` option to `true`.** See {% issue 687 %} for more details.
+{% endnote %}
+
+Use `cy.route2()` to manage the behavior of network requests at the network layer.
+
+## Comparison to cy.route()
+
+Unlike {% url `cy.route()` route %}, `cy.route2()`:
+
+- can mock all types of network requests (i.e., Fetch API, page loads, XMLHttpRequests, etc.)
+- no longer requires {% url `cy.server()` server %} before use
+
+# Syntax
+
+```javascript
+cy.route2(requestMatcher)
+cy.route2(requestMatcher, response)
+cy.route2(method, requestMatcher)
+cy.route2(method, requestMatcher, response)
+```
+
+## Usage
+
+**{% fa fa-check-circle green %} Correct Usage**
+
+```javascript
+cy.route2('/users/**')
+```
+
+## Arguments
+
+**{% fa fa-angle-right %} requestMatcher** ***(String, Glob, RegExp, Object)***
+
+The `requestMatcher` argument allows you to listen for a route matching a specific URL or pattern.
```

Comment:
The name of the type is `RouteMatcher`, not `RequestMatcher`:

https://github.com/cypress-io/cypress/blob/50a45c1b707ef0ccb0447b94ecb04946e994a4db/packages/net-stubbing/lib/external-types.ts#L209-L210

In general, I feel like the docs and types should be in line.

Reply:
@flotwig Agreed. My goal is at some point these should be automatically synced when the core codebase changed so it's autogenerated rather than manual syncing.